Add guidance for tables with lots of data (#3503)
alphagov/govuk-design-system: src/components/table/index.md

To learn more about this, read guidance on [extending and modifying components in production](/get-started/extending-and-modifying-components/).

{{ example({ group: "components", item: "table", example: "column-widths-custom-classes", html: true, nunjucks: true, open: false, size: "m" }) }}

## Tables with a lot of data

> If possible, you should aim to have less data in your tables. If you have a lot of data, try to organise it into multiple tables or mulltiple pages.

Reviewer: ```suggestion
If possible, you should aim to have less data in your tables. If you have a lot of data, try to organise it into multiple tables or multiple pages.
```

> If you can't split your data, you can use the CSS class `govuk-table--small-text-until-tablet`. This class reduces the size of the text on small screens so that the large amount of data is easier to read on smaller screens.

Reviewer: ```suggestion
If you cannot split your data, you can use the CSS class `govuk-table--small-text-until-tablet`. This class reduces the size of the text on small screens so the large amount of data is easier to read on smaller screens.
```

> If you cannot split your data, you can use the CSS class `govuk-table--small-text-until-tablet`. This class reduces the size of the text on small screens so that the large amount of data is easier to read on smaller screens.

Reviewer: ```suggestion
If you cannot split your data, you can use the CSS class `govuk-table--small-text-until-tablet`. This class reduces the size of text on small screens so the large amount of data is easier to read on smaller screens.
```
We can leave the last sentence as is, but I had a thought about the explanation. Would something like this work, or be too much detail?

'This class reduces the size of the text on small screens so the large amounts of data appears in longer sentences and retains its context.
In contrast, large text for large amounts of data can cause sentences to break over several lines making it harder to read on small screens.'

Author: I'm not 100% sure. I don't think the readability thing is about sentences breaking over multiple lines, it's about smaller text getting squashed together on a small view being easier to read than large text.

Good example is [this table on govuk](https://www.gov.uk/government/publications/budget-2020-documents/budget-2020#table-11-national-and-regional-employment-and-unemployment-rates-3-months-to-december-2019). It's a tricky example because currently it's set to 14px (😬) but if we increase the font size to 16px, it's still pretty ok on a small screen:

<img width="436" alt="Table with 16px font size table data" src="https://github.com/alphagov/govuk-design-system/assets/64783893/3b98c4cb-f8f1-46c0-b4a2-4e34e817cb9c">

If we increase it to 19px it starts to feel a bit crowded:

<img width="435" alt="Table with 19px font size table data" src="https://github.com/alphagov/govuk-design-system/assets/64783893/9ccae34d-6081-4103-9a64-5b7952353edc">

I'm struggling to articulate it clearly. I might need a hand from @christopherthomasdesign, if anything to make sure I've got this right. I think that explanation presents a nice tidy reason but I'm not sure it's the right reason.

Reviewer: OK, cool. Here's an alternative to consider with Chris:

'This class reduces the size of the text on small screens so large amounts of data has more empty space around it. This makes it easier to visually differentiate between each piece of data when read on small screens.'

Author: I like this tbh. Thanks Claire! I'll add it in and proceed. We can see what Chris says later and change it if needs be.

> You should not use this class on tables unless your table has a lot of data.

Reviewer 2: Would it make sense to be explicit about why?

Author: I've tried to add something to illustrate this. I'm not sure of the quality of the content. Lemme know what you think. CC @claireashworth.

> You should not use this class on tables unless your table has a lot of data because a smaller amount of data is easier to read if the text is larger.

Reviewer: ```suggestion
You should not use this class on tables unless your table has a lot of data, because a smaller amount of data is easier to read if the text is larger.
```
